src/OmniSharp.MSBuild/PackageDependencyChecker.cs

```diff
@@ -5,17 +5,55 @@
 using System.Linq;
 using Microsoft.Extensions.Logging;
 using NuGet.ProjectModel;
+using OmniSharp.Eventing;
+using OmniSharp.Models.Events;
 using OmniSharp.MSBuild.ProjectFile;
+using OmniSharp.Options;
+using OmniSharp.Services;
 
-namespace OmniSharp.MSBuild.Resolution
+namespace OmniSharp.MSBuild
 {
-    internal class PackageDependencyResolver
+    internal class PackageDependencyChecker
     {
         private readonly ILogger _logger;
+        private readonly IEventEmitter _eventEmitter;
+        private readonly DotNetCliService _dotNetCli;
+        private readonly MSBuildOptions _options;
 
-        public PackageDependencyResolver(ILoggerFactory loggerFactory)
+        public PackageDependencyChecker(ILoggerFactory loggerFactory, IEventEmitter eventEmitter, DotNetCliService dotNetCli, MSBuildOptions options)
```

Comment:
when we resolve inline package references (`#r "nuget: Foo, 1.0.0"`) in scripts, we create behind the scenes a temp `csproj` and restore using .NET CLI to resolve them. However, currently, we don't handle changes after project system init anymore. Looks like we'll be able to reuse this logic there 👍